src/librustc/middle/ty.rs

```diff
@@ -3181,21 +3295,20 @@ pub fn sequence_element_type<'tcx>(cx: &ctxt<'tcx>, ty: Ty<'tcx>) -> Ty<'tcx> {
     }
 }
 
-pub fn simd_type<'tcx>(cx: &ctxt<'tcx>, ty: Ty<'tcx>) -> Ty<'tcx> {
+pub fn simd_type<'tcx>(_cx: &ctxt<'tcx>, ty: Ty<'tcx>) -> Ty<'tcx> {
```

Comment:
This and a few other functions might benefit from not taking the unused `tcx` argument at all.